packages/core/src/types/asyncapi.ts

```diff
@@ -0,0 +1,1141 @@
+import { NodeType, listOf, mapOf } from '.';
+import { isMappingRef } from '../ref-utils';
+
+const Root: NodeType = {
+  properties: {
+    asyncapi: null, // TODO validate semver format and supported version
+    info: 'Info',
+    id: { type: 'string' },
+    servers: 'ServerMap',
+    channels: 'ChannelMap',
+    components: 'Components',
+    tags: 'TagList',
+    externalDocs: 'ExternalDocs',
+    defaultContentType: { type: 'string' },
+  },
+  required: ['asyncapi', 'channels', 'info'],
+};
+
+const Channel: NodeType = {
+  properties: {
+    description: { type: 'string' },
+    subscribe: 'Operation',
+    publish: 'Operation',
+    parameters: 'ParametersMap',
+    bindings: 'ChannelBindings',
+    servers: { type: 'array', items: { type: 'string' } },
+  },
+};
+
+const ChannelMap: NodeType = {
+  properties: {},
+  additionalProperties: 'Channel',
+};
+
+const ChannelBindings: NodeType = {
+  properties: {},
+  allowed() {
+    // allow all supported values, not all have deep linting
+    return [
+      'http',
+      'ws',
+      'kafka',
+      'anypointmq',
+      'amqp',
+      'amqp1',
+      'mqtt',
+      'mqtt5',
+      'nats',
+      'jms',
+      'sns',
+      'solace',
+      'sqs',
+      'stomp',
+      'redis',
+      'mercure',
+      'ibmmq',
+      'googlepubsub',
+      'pulsar',
+    ];
+  },
+  additionalProperties: { type: 'object' },
+};
+
+const Tag: NodeType = {
+  properties: {
+    name: { type: 'string' },
+    description: { type: 'string' },
+    externalDocs: 'ExternalDocs',
+  },
+  required: ['name'],
+};
+
+const ExternalDocs: NodeType = {
+  properties: {
+    description: { type: 'string' },
+    url: { type: 'string' },
+  },
+  required: ['url'],
+};
+
+const SecurityRequirement: NodeType = {
+  // done
```

Comment:
There are comments like this here and there. Please remove what is not needed anymore.

Reply:
Oops, left over from the draft version! Thanks for spotting, removed them.